hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientWriteHandler.java

```diff
@@ -60,17 +61,13 @@ public void handle() {
             ready = true;
             return;
         }
-        try {
-            writeBuffer();
-        } catch (Throwable t) {
-            logger.severe("Fatal Error at WriteHandler for endPoint: " + connection.getEndPoint(), t);
-        } finally {
```

Comment:
----> and consequent packets may be written safely.

THis is not going to work because you might have corrupted the stream. Some bytes may have been written to the bb and some have not.